Cpp_test/lib/ipc-handler.cpp

```diff
@@ -2,90 +2,150 @@
+	{
+		switch (opt)
+		{
+		case 'h':
+			info.protocol = IPCProtocol::HELP;
+			hflag = true;
+			std::cout << "<HELP>\n"
+						 "-q /<queue_name>, use queue as IPC method\n"
+						 "-p <pipe_name>, uses pipes as IPC method\n"
+						 "-s <shared_name> uses shared memory as IPC method\n"
+						 "-f <filename>, file used to read/write data\n"
+						 "Method name and file name can not start by '-'.\n"
+						 "Example: ./ipc_send -q /queue_method -f file_name\n"
+						 "</HELP>\n";
+			break;
+		case 'q':
+			qflag ++;
+			info.method_name = optarg;
+			break;
+		case 'p':
+			pflag ++;
+			info.method_name = optarg;
+			break;
+		case 's':
+			sflag ++;
+			info.method_name = optarg;
+			break;
+		case 'f':
+			fflag ++;
+			info.file_name = optarg;
 			break;
-			case 'q':
-				qflag = true; 
-				info.method_name = optarg;
-				break;
-			case 'p':
-				pflag = true;
-				info.method_name = optarg;
-				break;
-			case 's':
-				sflag = true;
-				info.method_name = optarg;
-				break;
-			case 'f':
-				fflag = true;
-				info.file_name = optarg;				
-				break;
-			default: /* '?' */
-				info.protocol = IPCProtocol::NONE;
-				throw std::runtime_error("Usage: \n"
-						"-h <--help print out containing description of all supported command-line arguments>\n");
+		default: /* '?' */
+			info.protocol = IPCProtocol::NONE;
+			throw std::runtime_error("Usage: \n"
+									 "-h <--help print out containing description of all supported command-line arguments>\n");
 		}
 	}
-  if (fflag) {
-		if ( !pflag && !qflag && !sflag) {
-			std::cout << "name of file is used: " << info.file_name << std::endl;
-			std::cout << "Select one IPC method!e.g \n"
-					"./ipcSend -q /<queue_name> -f <file_name>\n"
-					"./ipcSend -p <pipe_name> -f <file_name>\n"
-					"./ipcSend -s <shared_name> -f <file_name>\n";
-
-		} else {
-			if(pflag){
-				std::cout << "Pipe is used, pipe name is "<< info.method_name << std::endl;
-				std::cout << "File name: " << info.file_name << std::endl;
-				info.protocol = IPCProtocol::PIPE;	
-
-			} else if (qflag) {				
-				std::cout << "queue is used, queue name is "<< info.method_name << std::endl;
-				std::cout << "File name: " << info.file_name << std::endl;
-				info.protocol = IPCProtocol::QUEUE;
-
-			} else {
-				std::cout << "shared memory is used, shared name is "<< info.method_name << std::endl;
-				std::cout << "File name: " << info.file_name << std::endl;
-        		info.protocol = IPCProtocol::SHARE;
 
+	if (fflag)
+	{
+		if (pflag + qflag + sflag != 1)
+		{
+			if (info.file_name[0] == '-')
+			{
+				throw std::runtime_error("ERROR: File name can not start by '-', use -h for more information.\n");
 			}
-		}	
-
-	}else if (!hflag && (!pflag && !qflag && !sflag && !fflag)) {
+			else
+			{
+				std::cout << "name of file is used: " << info.file_name << std::endl;
+				std::cout << "Select one IPC method!e.g \n"
+							 "./ipcSend -q /<queue_name> -f <file_name>\n"
```

Comment:
Wrong program name in the example, same with two next lines.
```suggestion
							 "./ipc_send -q /<queue_name> -f <file_name>\n"
```